src/js/reducers.js

```diff
@@ -241,13 +240,16 @@ function deleteCommand(commands, cmdID) {
     return commands
 }
 function ui(state = {}, action) {
+    var newState = { ...state }
+    var app = newState[action.appID] ? newState[action.appID] : newAppState();
+    var menu = app.menu;
+    var menuItem = null;
+    var i = 0;
     switch (action.type) {
-        case Actions.SHOW:
-            var newState = { ...state }
-            var app = newState[action.appID] ? newState[action.appID] : newAppState()
+        case Actions.SHOW:           
             newState[action.appID] = app
```

Comment:
could `newState[action.appID] = app` be moved above the switch? it's in many cases. depends on mutability i guess,  if i change app later, does it change `newState[action.appID] ` (i think yes)

Reply:
Good idea, I think you are correct about the reference